component/prometheus/exporter/redis/redis.go

```diff
@@ -80,10 +86,27 @@ func (c *Config) UnmarshalRiver(f func(interface{}) error) error {
 	*c = DefaultConfig
 
 	type cfg Config
-	return f((*cfg)(c))
+	if err := f((*cfg)(c)); err != nil {
+		return err
+	}
+	return c.Validate()
+}
+
+func (c *Config) Validate() error {
+	if c.ScriptPath != "" && len(c.ScriptPaths) > 0 {
+		return fmt.Errorf("only one of script_path and script_paths should be specified")
```

Comment:
IMO, should implies it's optional like a warning, but it's being asserted here as a requirement:

```suggestion
		return fmt.Errorf("only one of script_path and script_paths can be specified")
```

On second read of this though, I'm not sure if "can" sounds any different than "should" so feel free to ignore this 🤷